app/models/osf-model.ts

```diff
@@ -272,4 +298,86 @@ export default class OsfModel extends Model {
             throw new Error(`Unexpected response ${errorContext}`);
         }
     }
+
+    /**
+     * Fetch one page of a has-many relationship using sparse fieldsets.
+     * See https://developer.osf.io/#tag/Sparse-Fieldsets
+     *
+     * The API response includes only the specified fields.
+     * This is useful for potentially long lists that require rendering only a few fields.
+     *
+     * Does NOT use ember-data. This means a few things:
+     * - Attributes don't pass through the transforms (e.g. 'fixstring') set with `DS.attr`, they remain as
+     *   represented in JSON. In particular, date fields are not deserialized to `Date` objects.
+     * - Sparse models aren't put in the store. This means no potential interactions with code that does use
+     *   the store, but if you want caching you'll have to do it yourself.
+     *
+     * Example:
+     * ```ts
+     * const contributors = await node.sparseHasMany(
+     *     'contributors',
+     *     { contributor: ['users'], user: ['fullName'] },
+     *     { queryParams: { 'page[size]': 100 } },
+     * });
+     *
+     * contributors.sparseModels.forEach(contrib => {
+     *     console.log(contrib.users.fullName);
+     * );
+     * ```
+     */
+    async sparseHasMany<T extends OsfModel>(
+        this: T,
+        relationshipName: RelationshipsFor<T>,
+        fieldset: SparseFieldset,
+        options: SparseHasManyOptions = {},
+    ): Promise<SparseHasManyResult> {
+        const response: ResourceCollectionDocument = await this.currentUser.authenticatedAJAX({
+            url: this.getHasManyLink(relationshipName),
+            data: {
+                fields: buildFieldsParam(fieldset),
+                ...options.queryParams,
+            },
+            ...options.ajaxOptions,
+        });
+
+        const { data, meta, links } = response;
+
+        return {
+            sparseModels: data.map(parseSparseResource),
+            meta,
+            ...(links ? { links } : {}),
+        };
+    }
+
+    /**
+     * Fetch the entirety of a has-many relationship using sparse fieldsets.
+     * See `sparseHasMany` above.
+     */
+    async sparseLoadAll<T extends OsfModel>(
+        this: T,
+        relationshipName: RelationshipsFor<T>,
+        fieldset: SparseFieldset,
+        options: SparseHasManyOptions = {},
+    ): Promise<SparseModel[]> {
+        const sparseModels: SparseModel[] = [];
+        let page = 1;
+        let totalPages = 0;
+
+        do { // eslint-disable-line no-await-in-loop
```

Comment:
Can this not be refactored into a Promise.all to take advantage of promises. (as the no-await-in-loop explains)

Reply:
We don't know how many pages there are until the first request comes back. I guess pages 2-X could be done with Promise.all, but with page size 100 that's a rare case that didn't seem worth optimizing for.

Reply:
Wait, so the 100 is hard coded in here? So the idea is just to loop to 100 each time? I'm also confused on why we can't wait for the request and do for(x bigger than pagelength) {gather promises} promise.all(gathered promises).

Reply:
Could have done it either way:
```
do {
  await ajax;
} while (more pages);
```
vs
```
await ajax;
await all(ajax for each remaining page)
```
I went with do-while because it felt nice to avoid writing the ajax call twice, and the two are equivalent when there are 2 or fewer pages of data. It's pretty rare that we have >200 things in a list, so I didn't worry about optimizing for >2 pages. Can change if you feel strongly about it, though.

Also, 100 is the largest page size the API allows. It's hard-coded several places, but should probably should be in the config. Maybe I'll make another PR with that, because it'd touch a bunch of unrelated things.

Reply:
hmm I'll leave it up to you, not a massive fan of do-whiles tbh but also don't feel that strongly about it.

Reply:
Drive by: 100 is the largest size the API allows by default. Some endpoints have that limit, and their default page size, set to 1000. That is generally as a way to get all records in one request and is only used for static data (like the list of institutions). It probably won't affect this discussion directly, but just in case.